domain/src/main/kotlin/no/nav/su/se/bakover/domain/søknadsbehandling/stønadsperiode/VerifisertStønadsperiodeOppMotPersonsAlder.kt

```diff
@@ -0,0 +1,101 @@
+import java.time.Clock
+import java.time.LocalDate
+import java.time.Year
+
+data class VerifisertStønadsperiodeOppMotPersonsAlder private constructor(
+    val stønadsperiode: Stønadsperiode,
+    val verifiseringsMelding: VerifiseringsMelding,
+) {
+    companion object {
+
+        @TestOnly
+        fun create(stønadsperiode: Stønadsperiode, verifiseringsMelding: VerifiseringsMelding) =
+            VerifisertStønadsperiodeOppMotPersonsAlder(stønadsperiode, verifiseringsMelding)
+
+        fun verifiser(
+            stønadsperiode: Stønadsperiode,
+            person: Person?,
+            clock: Clock,
+        ): Either<Valideringsfeil, VerifisertStønadsperiodeOppMotPersonsAlder> {
+            return when (person) {
+                null -> VerifisertStønadsperiodeOppMotPersonsAlder(
+                    stønadsperiode,
+                    VerifiseringsMelding.KunneIkkeVerifisereMotPerson,
+                ).right()
+
+                else -> verifiserMotPerson(stønadsperiode, person, clock)
+            }
+        }
+
+        private fun verifiserMotPerson(
+            stønadsperiode: Stønadsperiode,
+            person: Person,
+            clock: Clock,
+        ): Either<Valideringsfeil, VerifisertStønadsperiodeOppMotPersonsAlder> {
+            if (person.harFødselsInformasjon()) {
+                return verifiserMotFødselsinformasjon(person, stønadsperiode, clock)
+            }
+            return VerifisertStønadsperiodeOppMotPersonsAlder(
+                stønadsperiode,
+                VerifiseringsMelding.KunneIkkeVerifisereStønadsperiodeMotFødselsinformasjon,
+            ).right()
+        }
+
+        private fun verifiserMotFødselsinformasjon(
+            person: Person,
+            stønadsperiode: Stønadsperiode,
+            clock: Clock,
+        ): Either<Valideringsfeil, VerifisertStønadsperiodeOppMotPersonsAlder> {
+            val detteÅret = Year.now(clock)
+            if (person.eldreEnn67(detteÅret) == true) {
+                return Valideringsfeil.PersonEr67EllerEldre.left()
+            }
+            if (person.blir67(detteÅret) == true) {
+                if (person.harFødselsdato()) {
+                    return begrensStønadsperiodeVedBrukAvFødselsdato(person.fødsel?.dato!!, stønadsperiode).right()
+                }
+
+                return VerifisertStønadsperiodeOppMotPersonsAlder(
+                    stønadsperiode,
+                    VerifiseringsMelding.KunneIkkeVerifisereMotDetaljertFødselsinformasjon,
+                ).right()
+            }
+            return VerifisertStønadsperiodeOppMotPersonsAlder(
+                stønadsperiode,
+                VerifiseringsMelding.VerifisertOkPersonFyllerIkke67Plus,
+            ).right()
+        }
+
+        private fun begrensStønadsperiodeVedBrukAvFødselsdato(
+            dato: LocalDate,
+            stønadsperiode: Stønadsperiode,
+        ): VerifisertStønadsperiodeOppMotPersonsAlder {
+            val nyTilOgMed =
```

Comment:
Foreslår vi dropper å endre stønadsperioden for saksbehandleren, men istedet gir en left i det tilfellet vi vet at bruker fylte 67 i forrige måned eller tidligere. Enten vha. fødselsåret, eller datoen.